code-challenges.test.js

```diff
@@ -44,13 +115,56 @@ const letterE = "e"
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
-
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
+//    ReferenceError: findPairs is not defined
+
+//       123 |
+//       124 |
+//     > 125 |     expect(findPairs(hand1)).toEqual( true)
+//           |     ^
+//       126 |     expect(findPairs(hand2)).toEqual(false)
+//       127 |     expect(findPairs(hand3)).toEqual(false)
+//       128 |   })
+
+//       at Object.<anonymous> (code-challenges.test.js:125:5)
+
+// Test Suites: 1 failed, 1 total
+describe("findPairs", () => {
+  
+  it("takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.", () => {
+    const hand1 = [5, 5, 5, 3, 3]
+    // Expected output: true
+    const hand2 = [5, 5, 3, 3, 4]
+    // Expected output: false
+    const hand3 = [5, 5, 5, 5, 4]
+    // Expected output: false
+
+
+    expect(findPairs(hand1)).toEqual(true)
+    expect(findPairs(hand2)).toEqual(false)
+    expect(findPairs(hand3)).toEqual(false)
+  })
+})
+//1. Make a function 
+//2. Initially I was going to count the amount of pairs, you can have two pairs while and still be false, so im going to use the mode and the amount of pairs. If the mode, which is the most repeated number equals 5, and it still has to pairs, I believe I should still get a full house.
+// 3. I'll need an iterator, variable for mode, and a variable for pairs. 
+
+
+const findPairs = (arr)=>{
+    let counter={};
+    for(let i =0;i<arr.length;i++){
+        counter[arr[i]] = counter[arr[i]]+1 ||0;
+    }
+    var pairCount = Object.keys(counter).reduce((acc,curr)=>{
+      acc += Math.floor(curr/2)
+      return acc;
+    },0);
+    let mode = Object.keys(counter).reduce((a,b)=>counter[a]> counter[b]? a:b);
+    if( pairCount==3&& mode==5){
+      return true;
+    }else{
+      return false;
+    }
+}
```

Comment:
Wow, this is a really unique approach, but it looks like it works!  What made you decide to use objects inside this function as the counter?